### PR TITLE
Collectives integration tests xcm v3

### PR DESCRIPTION
**Pull request by muharem**
Adopting existing integration tests of the Polkadot Collectives chain for xcm v3.

Requires:
https://github.com/paritytech/parachains-integration-tests/pull/89
https://github.com/paritytech/cumulus/pull/2222

### CODE REVIEW

File: parachains/integration-tests/collectives/0_xcm/2_reserve.yml [hunk 1]
```diff
@@ -42,43 +40,35 @@ tests:
...
-              ]
-              events:
-                - name: system.ExtrinsicFailed
-                  attributes:
-                    - type: SpRuntimeDispatchError
-                      key: dispatchError
-                      # TODO assert variant
-                      # issue - https://github.com/paritytech/parachains-integration-tests/issues/59
-                      value: {"Module":{"index":"31","error":"0x02000000"}}
+      # Fails with the message - "RpcError: 1010: Invalid Transaction: Transaction would exhaust the block limits"
```
Comment (bkontur) [inline parachains/integration-tests/collectives/0_xcm/2_reserve.yml:51]: @muharem
does this have any following issue or solution or it just stays commented out?

**Review by bkontur — APPROVED** [review]

File: parachains/integration-tests/collectives/0_xcm/2_reserve.yml (hunk 1 above)
Comment (muharem) [reply parachains/integration-tests/collectives/0_xcm/2_reserve.yml:51]: we need parachains-integration-tests be able to handle it 
but better remove it for now

Comment (muharem) [reply parachains/integration-tests/collectives/0_xcm/2_reserve.yml]: removed

**Review by gilescope — APPROVED** [review]
Looks good.

File: parachains/integration-tests/collectives/0_xcm/0_init.yml [hunk 2]
```diff
@@ -62,13 +62,26 @@ tests:
...
+                        }
+                      }
+                    },
                     {
                       Transact: {
-                        originType: Superuser,
-                        requireWeightAtMost: 1000000000, # 1_000_000_000
+                        originKind: Superuser,
+                        requireWeightAtMost: {
+                          refTime: 200000000, # 200_000_000
```
Comment (ruseinov) [inline parachains/integration-tests/collectives/0_xcm/0_init.yml:82]: Curious where this values are coming from, seems different to what it used to be, but also the weight format has changed.

File: parachains/integration-tests/collectives/0_xcm/0_init.yml [hunk 3]
```diff
@@ -62,13 +62,26 @@ tests:
...
-                { v1: { 0, interior: { x1: { parachain: *cp_id }}}}, # destination
+                { v3: { 0, interior: { x1: { parachain: *cp_id }}}}, # destination
                 {
-                  v2: [ # message
+                  v3: [ # message
+                    {
+                      UnpaidExecution: {
+                        weightLimit: { 
+                          limited: {
+                            refTime: 2200000000, # 2_200_000_000
```
Comment (ruseinov) [inline parachains/integration-tests/collectives/0_xcm/0_init.yml:72]: same here, would nice to understand the logic behind weight values.

File: parachains/integration-tests/collectives/1_alliance/1_init_alliance.yml [hunk 4]
```diff
@@ -159,13 +181,26 @@ tests:
...
+                        }
+                      }
+                    },
                     {
                       Transact: {
-                        originType: Superuser,
-                        requireWeightAtMost: 100000000000, # 100_000_000_000
+                        originKind: Superuser,
+                        requireWeightAtMost: {
+                          refTime: 3000000000, # 1_000_000_000 2_440_115_783
```
Comment (ruseinov) [inline parachains/integration-tests/collectives/1_alliance/1_init_alliance.yml:201]: Trying to understand what these comments next to weights mean. As far as I can see we used to just have a more readable version `100000000000` -> `100_000_000_000`, but now there is more info. Some of it seems to be something that is coming up upon test execution, some of it is something else. Would great to have clarity here.

File: parachains/integration-tests/collectives/0_xcm/2_reserve.yml [hunk 5]
```diff
@@ -42,43 +40,10 @@ tests:
...
                 - name: dmpQueue.ExecutedDownward
                   chain: *collectives_parachain
                   attributes:
-                    - type: XcmV2TraitsOutcome
+                    - type: XcmV3TraitsOutcome
                       xcmOutcome: Incomplete
-                      # the weight must be static
-                      value: [1000000000, UntrustedReserveLocation]
-
-      - name: Reserve assets from Collectives Parachain to Relay Chain fails
```
Comment (ruseinov) [inline parachains/integration-tests/collectives/0_xcm/2_reserve.yml:57]: Why is this test no longer needed?

**Comment by ruseinov** [comment]
Overall looks good, got some questions though. 

Prior to merging this it would be nice to get some clarity on:
1. Removed test(s), if they need to be reinstated later - an issue and a comment would be nice
2. Changed weights
3. Comments next to the new weights

I'm guessing once this gets merged it also need to make it's way to the `integration-tests` repo, which is what we are actually executing to run those, right?

File: parachains/integration-tests/collectives/1_alliance/3_kick_member.yml [hunk 6]
```diff
@@ -129,13 +120,26 @@ tests:
...
-                { v1: { parents: 0, interior: { x1: { parachain: *cp_id }}}}, # destination
+                { v3: { parents: 0, interior: { x1: { parachain: *cp_id }}}}, # destination
                 {
-                  v2: [ #message
+                  v3: [ #message
+                    {
+                      UnpaidExecution: {
+                        weightLimit: { 
+                          limited: {
+                            refTime: 4000000000, # 3_000_000_000
```
Comment (ruseinov) [inline parachains/integration-tests/collectives/1_alliance/3_kick_member.yml:130]: The comment here seems inconsistent with the value

**Comment by NachoPal** [comment]
> I'm guessing once this gets merged it also need to make it's way to the `integration-tests` repo, which is what we are actually executing to run those, right?

The tests were meant to be run in this repo:
https://github.com/paritytech/ci_cd/issues/499
https://github.com/paritytech/cumulus/pull/1678

Unfortunately, it never made his way to `master` though. It was actually working. Please catch up with @alvicsam to see if we can merged.

File: parachains/integration-tests/collectives/0_xcm/0_init.yml (hunk 3 above)
Comment (muharem) [reply parachains/integration-tests/collectives/0_xcm/0_init.yml:72]: so this just limit, use at most. 
1_000_000_000 per xcm instruction plus 200_000_000 for call 
not trying to make it precise or test anything related to weigths.

File: parachains/integration-tests/collectives/0_xcm/2_reserve.yml (hunk 5 above)
Comment (muharem) [reply parachains/integration-tests/collectives/0_xcm/2_reserve.yml:57]: here the answer https://github.com/paritytech/cumulus/pull/2221/commits/5f3fda69b0352fcbdcdd3192f14c6a20a9bf1b2d

File: parachains/integration-tests/collectives/1_alliance/1_init_alliance.yml (hunk 4 above)
Comment (muharem) [reply parachains/integration-tests/collectives/1_alliance/1_init_alliance.yml:201]: my debugging comments, I remove them and leave only readably version

**Comment by muharem** [comment]
@alvicsam please advise

File: parachains/integration-tests/collectives/1_alliance/3_kick_member.yml (hunk 6 above)
Comment (gilescope) [reply parachains/integration-tests/collectives/1_alliance/3_kick_member.yml:130]: think we can ditch these comments - they don't add that much.

**Comment by alvicsam** [comment]
The last time we stopped because the tests didn't work (unfortunately logs are gone). I can try one more time to check if now they can work.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2844101

**Comment by muharem** [comment]
bot rebase

**Comment by muharem** [comment]
bot merge

**Pull request merged**